### PR TITLE
Added support for .NET Core

### PR DESCRIPTION
**Pull request by santriseus**
Moved to a Class Library (Package) project type.
Add support for .NET Core using compiler directives.
Created a stub library with rerefences to existing code files,  for testing, since it is not possible to reference the .xproj project directly from Tests project.
Output nuget packages for publishing will be located in the SolutionDir/artifacts

### CODE REVIEW

**Comment by santriseus** [comment]
Unfortunately support for vNext and .NET Core requires completely different way of building the library.
Even if you add the coreclr50 folder to the nuget package you will receive the error during the compilarion time, since you need to elliminate dependance to the original mscorlib :( 
So I`ve added this new project, which should buid the sources for different supported platforms.
Tested it in VS 2015. Please, take a look at my changes, and tell me if something does not work for you.
Also since resharper lib does not support coreclr i had to put it in conditional processor directive.

File: AsyncIO/FileSystem/AsyncFile.cs [hunk 1]
```diff
@@ -146,8 +156,10 @@ public static Task MoveAsync(string sourceFileName, string destFileName)
             PathValidator.EnsureCorrectFileSystemPath(sourceFileName);
             PathValidator.EnsureCorrectFileSystemPath(destFileName);
 
+#if !DNXCORE50
             if (Path.GetFullPath(sourceFileName) == Path.GetFullPath(destFileName))
                 return;
+#endif
```
Comment (FireNero) [inline AsyncIO/FileSystem/AsyncFile.cs]: Can be replaced by following code without compiler directive: 
            if (Path.Combine(Path.GetDirectoryName(sourceFileName), sourceFileName) == Path.Combine(Path.GetDirectoryName(destFileName), destFileName))
                return;

File: AsyncIO/FileSystem/AsyncFile.cs [hunk 2]
```diff
@@ -146,8 +156,10 @@ public static Task MoveAsync(string sourceFileName, string destFileName)
             PathValidator.EnsureCorrectFileSystemPath(sourceFileName);
             PathValidator.EnsureCorrectFileSystemPath(destFileName);
 
+#if !DNXCORE50
```
Comment (FireNero) [inline AsyncIO/FileSystem/AsyncFile.cs]: Can be replaced by following code without compiler directive: 
            if (Path.Combine(Path.GetDirectoryName(sourceFileName), sourceFileName) == Path.Combine(Path.GetDirectoryName(destFileName), destFileName))
                return;

Comment (FireNero) [inline AsyncIO/FileSystem/AsyncFile.cs]: 159

Comment (firenero) [inline AsyncIO/FileSystem/AsyncFile.cs]: Can be replaced by following code without compiler directive: 
            if (Path.Combine(Path.GetDirectoryName(sourceFileName), sourceFileName) == Path.Combine(Path.GetDirectoryName(destFileName), destFileName))
                return;

**Comment by firenero** [comment]
Your solution is working ok for me. 
I've also found solution how to avoid compiler directives. You can update JetBrains.Annotations package to 10.1.1-eap version that supports .net core. Here it is https://www.nuget.org/packages/JetBrains.Annotations/10.1.1-eap
I think that will keep code more clear.

Please apply this changes to your solution and I'll gladly merge your pull request.
Thanks.

**Pull request merged**